view.c

```diff
@@ -234,16 +111,11 @@ view_map(struct cg_view *view, struct wlr_surface *surface)
 {
 	view->wlr_surface = surface;
 
-	struct wlr_subsurface *subsurface;
-	wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces_below, parent_link) {
-		subsurface_create(view, subsurface);
-	}
-	wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces_above, parent_link) {
-		subsurface_create(view, subsurface);
+	view->scene_node = wlr_scene_subsurface_tree_create(&view->server->scene->node, surface);
+	if (!view->scene_node) {
+		return;
```

Comment:
Same here as with the drag icons: I think we need to do some accounting here before returning early.